ir/src/symbol_table.rs

```diff
@@ -151,6 +159,23 @@ impl SymbolTable {
         Ok(())
     }
 
+    pub(super) fn insert_random_values(
+        &mut self,
+        values: &RandomValues,
+    ) -> Result<(), SemanticError> {
+        // TODO: add name of random values vector to the identifiers. For now it can be called only
+        // by `$rand`
+        if values.size() > 0 {
+            self.random_values_num = values.size() as u16;
+        } else {
+            self.random_values_num = values.values().len() as u16;
```

Comment:
The logic for figuring out the number of random values should probably be encapsulated in the `RandomValues` struct.